rust/ql/lib/codeql/rust/elements/internal/MethodCallExprImpl.qll

```diff
@@ -43,10 +43,15 @@ module Impl {
     }
 
     override string toStringImpl() {
-      exists(string base, string separator |
+      exists(string base, string separator, string parenthesis |
         base = this.getReceiver().toAbbreviatedString() and
         (if base = "..." then separator = " ." else separator = ".") and
-        result = base + separator + this.getIdentifier().toStringImpl() + "(...)"
+        (
+          if this.getArgList().getNumberOfArgs() = 0
+          then parenthesis = "()"
+          else parenthesis = "(...)"
+        ) and
+        result = base + separator + this.getIdentifier().toStringImpl() + parenthesis
```

Comment:
It would be better to use the `toStringPart` pattern here instead, to avoid computing intermediate strings